howto/stream/exchange-oob-data.md

```diff
@@ -2,127 +2,151 @@
 
-Create a data channel (named `foo` in the following example) under the `dataChannels` property of an `AnboxStream` object and register event handlers that respond to the events sent from an Android application:
+#### Extend `AnboxStream` Configuration
+
+Now, to extend an `AnboxStream` object, add the `dataChannels` property to define a new data channel. For example:
 
 ```
-let stream = new AnboxStream({
-  ...
-  ...
-  dataChannels: {
-    "foo": {
-      callbacks: {
-        close: () => {
-          console.log('data channel is closed')
-        },
-        open: () => {
-          console.log('data channel is open')
-        },
-        error: (err) => {
-          console.log(`error: ${err}`)
-        },
-        message: (data) => {
-          console.log(`data received: ${data}`)
-        }
-      }
-    }
-  }
-});
+  <body>
+    <div id="anbox-stream"></div>
+    <div id="controls" style="position: absolute; top: 20px; left: 20px; z-index: 10">
+      <input type="text" id="textBox" placeholder="Send message">
+      <button id="sendButton">Send</button>
+    </div>
+
+    <script>
+      ...
+      window.onload = () => {
+        const stream = new AnboxStream({
+          targetElement: "anbox-stream",
+          ...
+          dataChannels: {
+            "foo": {
+              callbacks: {
+                close: () => {
+                  console.log('data channel is closed')
+                },
+                open: () => {
+                  console.log('data channel is open')
+                },
+                error: (err) => {
+                  console.log(`error: ${err}`)
+                },
+                message: (data) => {
+                  console.log(`data received: ${data}`)
+                }
+              }
+            }
+          }
+        });
+        stream.connect();
+
+        const sendButton = document.getElementById("sendButton");
+        const textBox = document.getElementById("textBox");
+        sendButton.addEventListener("click", () => {
+          const message = textBox.value;
+          if (message) {
+            stream.sendData('foo', message);
+          }
+        });
+      };
+    </script>
+  </body>
+
 ```
 
+1. This extends a `AnboxStream` object to create a data channel named `foo` and registers event handlers for data communication between the Anbox instance and web client.
+2. It also adds a new section to the web page, containing an input field to type and send a message to Anbox runtime through the created data channel.
+
+When joining an existing stream-enabled instance, the data channel can be created dynamically based on the specific configuration of the instance.
```

Comment:
I am not sure I can follow here. What do you mean with "the specific configuration of the instance"?